controllers/pxc/transition_key_copier.go

```diff
@@ -0,0 +1,106 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pxcv1 "github.com/Percona-Lab/k8s-vault-issuer/apis/pxc/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *PerconaXtraDBClusterReconciler) processTransferVaultKeys(currClusterCR *pxcv1.PerconaXtraDBCluster, clusters []string) error {
+	currClusterName := currClusterCR.Name + "." + currClusterCR.Namespace
+	for _, v := range clusters {
+		splittedName := strings.Split(v, ".")
+		if len(splittedName) != 2 {
+			return errors.New("Invalid source cluster name, plaease use format clusterName.namespace")
+		}
+
+		clusterName, clusterNamespace := splittedName[0], splittedName[1]
+
+		srcClusterCR := &pxcv1.PerconaXtraDBCluster{}
+		err := r.Client.Get(context.TODO(),
+			types.NamespacedName{
+				Namespace: clusterNamespace,
+				Name:      clusterName,
+			}, srcClusterCR)
+		if err != nil {
+			return errors.Wrap(err, "get cluster srcClusterCR from namespace "+v)
+		}
+
+		values, ok := srcClusterCR.Annotations["percona.com/allow-transition-key-transfer"]
+		if !ok {
+			return fmt.Errorf("No percona.com/allow-transition-key-transfer annotation found at cluster %s", v)
+		}
+
+		allowed := false
+		for _, allowedClusters := range strings.Split(values, ",") {
+			if allowedClusters == currClusterName {
+				allowed = true
```

Comment:
```
if err := r.transferTransitionKeys(srcClusterCR, currClusterCR); err != nil {
			return errors.Wrapf(err, "transfer transition key from:%s to:%s", v, currClusterCR.Namespace)
}
```

Reply:
?
tranferTransitionKeys call is inside for loop, while deleteAnnotation is outside, since it's deleting annotation from destination cluster in the end

Reply:
Oh, yes, then just do
```
if err := r.transferTransitionKeys(srcClusterCR, currClusterCR); err != nil {
	return errors.Wrapf(err, "transfer transition key from:%s to:%s", v, currClusterCR.Namespace)
}
break
```
And I think it's not good to return error here because you interrupt the loop and do not iterate over next items in `cluster` slice.
So if you have at least 1 not allowed cluster in annotation this code will try to do it until it's there and will not be able to handle keys from allowed clusters.